src/main/java/com/percero/agents/sync/access/RedisAccessManager.java

```diff
@@ -1327,14 +1397,22 @@ public void checkChangeWatchers(ClassIDPair classIdPair, String[] fieldNames, St
 		Set<String> fieldWatcherKeys = new HashSet<String>();
 		
 		String objectWatchedFieldKey = RedisKeyUtils.objectWatchedFields(classIdPair.getClassName(), classIdPair.getID());
-		Set<Object> objectWatcherValueKeys = redisDataStore.getHashKeys(objectWatchedFieldKey);
-		Iterator<Object> itrObjectWatcherValueKeys = objectWatcherValueKeys.iterator();
-		while (itrObjectWatcherValueKeys.hasNext()) {
-			String nextObjectWatcherValueKey = (String) itrObjectWatcherValueKeys.next();
+		// Also add for ALL objects of this class type
+		String allObjectsWatchedFieldKey = RedisKeyUtils.objectWatchedFields(classIdPair.getClassName(), "0");
 
-			// Get all fields for this object that are being watched and remove them.
-			String changeWatcherKey = (String) redisDataStore.getHashValue(objectWatchedFieldKey, nextObjectWatcherValueKey);
-			fieldWatcherKeys.add(changeWatcherKey);
+		String[] fieldKeys = {objectWatchedFieldKey, allObjectsWatchedFieldKey};
```

Comment:
I'm not really able to follow what the change watcher changes you've made are intended to do. If you want a useful code review of that stuff we'll have to do it together.